websites/A/Animes Portal/presence.ts

```diff
@@ -0,0 +1,365 @@
+const presence = new Presence({
+  clientId: "924791712944099389"
+});
+
+const filterCodes: Record<number, string> = {
+  366: "BG sub"
+};
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+
+  const pathname: string = document.location.pathname;
+  const paths = pathname.split("/");
+  if (!paths[0]) paths.shift();
+
+  if (pathname === "/") {
+    presenceData.details = "Home";
+    presenceData.state = "Viewing the home page";
+  } else if (pathname.startsWith("/messages")) {
+    if (!paths[1]) {
+      presenceData.details = "Viewing messages";
+    } else if (paths[1].startsWith("pm-")) {
+      const body = document.querySelector(
+        "body > main.animated > div.wrapper > div.dialogPadding"
+      );
+      const username = body.querySelector("span.username").textContent;
+      const avatar = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Messaging ${username}`;
+      presenceData.smallImageKey = avatar;
+      presenceData.smallImageText = `Messaging ${username}`;
+    }
+  } else if (pathname === "/chat") {
+    presenceData.details = "Chatting in the chat";
```

Comment:
More appropriate string?

Reply:
```suggestion
    presenceData.details = "Chatting";
```

Reply:
agreed

Reply:
See, there is messages in the chat, and one global chat for everyone in the site